engine/crates/engine-v2/src/execution/response_modifier.rs

```diff
@@ -0,0 +1,109 @@
+use std::sync::Arc;
+
+use itertools::Itertools;
+
+use crate::{
+    operation::ResponseModifierRule,
+    response::{ErrorCode, GraphqlError, InputdResponseObjectSet, ResponseBuilder, UnpackedResponseEdge},
+    Runtime,
+};
+
+use super::{state::OperationExecutionState, ExecutionContext, ResponseModifierExecutorId};
+
+impl<'ctx, R: Runtime> ExecutionContext<'ctx, R> {
+    pub(super) async fn execute_response_modifier(
+        &self,
+        state: &mut OperationExecutionState<'ctx>,
+        response: &mut ResponseBuilder,
+        response_modifier_executor_id: ResponseModifierExecutorId,
+    ) {
+        let executor = &self.operation[response_modifier_executor_id];
```

Comment:
This is very confusing.  Why is a thing named context doing the execution, and the thing named executor not doing the one thing that it's name says it does.

I'm not even sure what to suggest here - is the executor named wrong?  Would this make more sense as a function on the executor, or just a freestanding function that _takes_ a context?

Reply:
Oh, good catch. Yeah, this is more of a freestanding function. You're right that it should be an executor method for clarity, I mostly started with it not entirely how it would end up and left it as is